test/psim/core/counter.hpp

```diff
@@ -0,0 +1,30 @@
+/** @file test/psim/core/counter.hpp
+ *  @author Kyle Krol
+ */
+
+#ifndef TEST_PSIM_CORE_COUNTER_HPP_
+#define TEST_PSIM_CORE_COUNTER_HPP_
+
+#include <psim/core/configuration.hpp>
+#include <psim/core/model.hpp>
+#include <psim/core/parameter.hpp>
+#include <psim/core/state_field_valued.hpp>
+#include <psim/core/types.hpp>
+
+/** @brief Simple model that counts an integer forward for testing purposes.
```

Comment:
If this is only used for testing purposes why keep it out in core/?

Reply:
Not sure if I understand the question here? It's only for testing core infrastructure so I just left it in `test/psim/core`?

Reply:
Oh I'm dumb lol. I didn't see the test/ at the beginning.